tests/scripts/pulpcore/test_repository_reclaim_space.sh

```diff
@@ -0,0 +1,46 @@
+#!/bin/bash
+
+# shellcheck source=tests/scripts/config.source
+. "$(dirname "$(dirname "$(realpath "$0")")")"/config.source
+
+pulp debug has-plugin --name "core" --min-version "3.19.0" || exit 23
+pulp debug has-plugin --name "file" || exit 23
+
+cleanup() {
+  pulp file repository destroy --name "cli_test_reclaim_repo" || true
+  pulp file remote destroy --name "cli_test_reclaim_remote" || true
+  pulp file distribution destroy --name "cli_test_reclaim_distro" || true
+}
+trap cleanup EXIT
+
+# On demand policy ensures that the artifacts are not downloaded until it is requested
+expect_succ pulp file remote create --name "cli_test_reclaim_remote" --url "$FILE_REMOTE_URL" --policy on_demand
+expect_succ pulp file repository create --name "cli_test_reclaim_repo" --remote "cli_test_reclaim_remote" --autopublish
+expect_succ pulp file repository sync --name "cli_test_reclaim_repo"
+expect_succ pulp file distribution create --name "cli_test_reclaim_distro" --base-path "cli_test_reclaim_distro" --repository "cli_test_reclaim_repo"
+DIST_BASE_URL="$(echo "$OUTPUT" | jq -r '.base_url')"
+
+# Download the artifact and verify that it exists
+expect_succ curl "${DIST_BASE_URL}1.iso"
```

Comment:
make the sync immediate instead?

Reply:
I though the reclaim space command is mostly useful for cases where the content is downloaded on demand. The content gets stored when user requests it, then it can be deleted via reclaim-space, and then the user can download it again with the on_demand policy.

Reply:
It is useful in any case where the content has been downloaded onto the system. You can perform an immediate sync and then turn the repo into an on-demand repo by using reclaim. When Pulp syncs content, immediate or on-demand we remember where we got the content from so we can download it again in the future.